kube-controllers/tests/fv/metrics_test.go

```diff
@@ -318,88 +394,129 @@ var _ = Describe("kube-controllers metrics tests", func() {
 		// Assert that IPAM metrics have been updated. We should no longer have any allocations on NodeB, however
 		// the block should still exist. Since we don't release affinity until the block is empty, the block should
 		// still be affine to node-b.
-		notExpectedMetrics := []string{
-			`ipam_allocations_per_node{node="node-b"}`,
-			`ipam_blocks_per_node{node="node-c"}`,
-		}
-		expectedMetrics = []string{
-			`ipam_allocations_per_node{node="node-a"} 4`,
-			`ipam_allocations_per_node{node="node-c"} 2`,
-			`ipam_blocks_per_node{node="node-a"} 1`,
-			`ipam_blocks_per_node{node="node-b"} 1`,
-			`ipam_blocks_per_node{node="no_affinity"} 1`,
-			`ipam_allocations_borrowed_per_node{node="node-c"} 2`,
-		}
-		Eventually(func() error {
-			out, err = getMetrics(fmt.Sprintf("http://%s:9094/metrics", kubeControllers.IP))
-			Expect(err).NotTo(HaveOccurred())
-			for _, s := range expectedMetrics {
-				if !strings.Contains(out, s) {
-					return fmt.Errorf("Expected:\n%s\nTo contain metric:\n  %s\n", out, s)
-				}
-			}
-			for _, s := range notExpectedMetrics {
-				if strings.Contains(out, s) {
-					return fmt.Errorf("Expected:\n%s\nNOT to contain metric:\n  %s\n", out, s)
-				}
-			}
-			return nil
-		}, 5*time.Second).ShouldNot(HaveOccurred())
+		validateExpectedAndUnexpectedMetrics(
+			[]string{
+				`ipam_allocations_in_use{ippool="test-ippool-1",node="node-a"} 4`,
+				`ipam_allocations_in_use{ippool="no_ippool",node="node-a"} 1`,
+				`ipam_allocations_in_use{ippool="test-ippool-1",node="node-c"} 2`,
+				`ipam_allocations_per_node{node="node-a"} 5`,
+				`ipam_allocations_per_node{node="node-c"} 2`,
+				`ipam_blocks{ippool="test-ippool-1",node="node-a"} 1`,
+				`ipam_blocks{ippool="test-ippool-1",node="node-b"} 1`,
+				`ipam_blocks{ippool="no_ippool",node="no_affinity"} 1`,
+				`ipam_blocks{ippool="test-ippool-1",node="no_affinity"} 1`,
+				`ipam_blocks_per_node{node="node-a"} 1`,
+				`ipam_blocks_per_node{node="node-b"} 1`,
+				`ipam_blocks_per_node{node="no_affinity"} 2`,
+				`ipam_allocations_borrowed{ippool="test-ippool-1",node="node-c"} 2`,
+				`ipam_allocations_borrowed_per_node{node="node-c"} 2`,
+			},
+			[]string{
+				`ipam_allocations_gc_reclamations{ippool="test-ippool-1",node="node-b"}`,
```

Comment:
Do we expect to see any reclamations? Maybe we should just assert on metric without the labels to ensure there are none at all.